packages/compile-solidity/src/compilerSupplier/Cache.ts

```diff
@@ -24,13 +26,31 @@ export class Cache {
   add(code: string, fileName: string) {
     const filePath = this.resolve(fileName);
     fs.writeFileSync(filePath, code);
+    this.memoizedCompilers[fileName] = code;
   }
 
   has(fileName: string) {
     const file = this.resolve(fileName);
     return fs.existsSync(file);
   }
 
+  loadFile(fileName: string): string {
+    if (this.memoizedCompilers.has(fileName)) {
+      return this.memoizedCompilers[fileName];
```

Comment:
Same consideration as above, the key should be the resolved path

Reply:
I'll change it but currently it doesn't look like two files can have the same name as only one directory is used to cache the compilers